### PR TITLE
Deprecate imread() reading from URLs

### LINKED ISSUE #18648: Drop support for directly imread()ing urls.
### Describe the issue

**Summary**

Support for directly imread()ing url strings caused Matplotlib to add a dependency on a package which we should not have no business of having a dependency on (`certifi`, see https://github.com/matplotlib/matplotlib/pull/18636#issuecomment-703126749).  url strings can't be directly `open`'ed either, so if we instead documented the (easy) solution of calling `urllib.request.urlopen` on the URL (+/- certifi) it'd be much more educative for users (now they know how to urlopen anything, not just images) (note that one of the main reasons for originally adding support to directly opening url strings was that the name of `urllib.request.urlopen` changed between Py2 and Py3, https://github.com/matplotlib/matplotlib/pull/4256#issuecomment-84369830, but that's obviously not relevant anymore).  Finally if we really want that functionality it would make much more sense to have it in Pillow anyways (given that imread is just a thin wrapper around PIL.Image.open at that point).

### Proposed fix

Deprecate and then drop support for `imread("https://...")`.

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Closes #18648.

### CODE REVIEW

**Review by dopplershift — APPROVED** [review]

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
recyclying to get doc build to rerun...

File: doc/api/next_api_changes/deprecations/18649-TH.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+``imread()`` reading from URLs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Passing a URL to `~.pyplot.imread()` is deprecated. Please open the URL before
+reading using ``urllib.request.urlopen()``.
```
Comment (QuLogic) [inline doc/api/next_api_changes/deprecations/18649-TH.rst:5]: As part of the standard library, shouldn't this be link-able?

**Comment by tacaswell** [comment]
The user also has to do a bit more jiggery-pokery than just `urlopen` as pillow expects to be able to seek on the buffer and what comes back from urlopen does not support seek (see discussion in https://github.com/matplotlib/matplotlib/issues/18129). 

The documented incantation needs to be `image.imread(io.BytesIO(urllib.request.urlopen(url).read()))` or `image.imread(io.BytesIO(requests.get(url).content))`.

I suspect we should also abandon most of the png guessing code and just let pillow infer as much as it wants to.  I think this would make our code simpler (as it is almost a completed drop-through to pillow) and make it more functional (it will open things other than png from a url).

 I am :+1: on deprecating this, but we need to provide more guidance about what to replace it with.

**Review by tacaswell — CHANGES_REQUESTED** [review]
Needs better docs on path away from imread.

Should probably also have, "Use `PIL.image.open(urlopen(url))`" as an option too.

**Comment by timhoffm** [comment]
:+1: on the suggestions.

Anybody, feel free to amend. I'm not in word-picking mood right now.

**Comment by jklymak** [comment]
close in favour of #19367

**Pull request closed without merging**